modules/createTransitionManager.js

```diff
@@ -226,15 +210,6 @@ export default function createTransitionManager(history, routes) {
         if (history.listenBeforeUnload)
           unlistenBeforeUnload = history.listenBeforeUnload(beforeUnloadHook)
       }
-    } else {
-      if (hooks.indexOf(hook) === -1) {
```

Comment:
Ah shit, I was racing through this to get back to work. I'll have to look at it tonight.